tests/test_evpn_tunnel.py

```diff
@@ -0,0 +1,749 @@
+        self.tunnel_term_ids = get_exist_entries(dvs, self.ASIC_TUNNEL_TERM_ENTRY)
+        self.bridgeport_ids = get_exist_entries(dvs, self.ASIC_BRIDGE_PORT)
+
+    def check_vxlan_tunnel_map_entry_delete(self, dvs, tunnel_name, vidlist, vnilist):
+        asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+
+        tbl = swsscommon.Table(asic_db, self.ASIC_TUNNEL_MAP_ENTRY)
+
+        for x in range(len(vidlist)):
+            status, fvs = tbl.get(self.mapentry_map[tunnel_name + vidlist[x]])
+            assert status == False, "SIP Tunnel Map entry not deleted"
+            iplinkcmd = "ip link show type vxlan dev " + tunnel_name + "-" + vidlist[x]
+            (exitcode, out) = dvs.runcmd(iplinkcmd)
+            assert exitcode != 0, "Kernel device not deleted"
+
+    def check_vxlan_tunnel_map_entry(self, dvs, tunnel_name, vidlist, vnidlist):
+        asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+
+        tbl = swsscommon.Table(asic_db, self.ASIC_TUNNEL_MAP_ENTRY)
+
+        expected_attributes_1 = {
+        'SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP_TYPE': 'SAI_TUNNEL_MAP_TYPE_VNI_TO_VLAN_ID',
+        'SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP': self.tunnel_map_map[tunnel_name][0],
+        'SAI_TUNNEL_MAP_ENTRY_ATTR_VLAN_ID_VALUE': vidlist[0],
+        'SAI_TUNNEL_MAP_ENTRY_ATTR_VNI_ID_KEY': vnidlist[0],
+        }
+
+        for x in range(len(vidlist)):
+            expected_attributes_1['SAI_TUNNEL_MAP_ENTRY_ATTR_VLAN_ID_VALUE'] = vidlist[x]
+            expected_attributes_1['SAI_TUNNEL_MAP_ENTRY_ATTR_VNI_ID_KEY'] = vnidlist[x]
+            ret = get_key_with_attr(asic_db, self.ASIC_TUNNEL_MAP_ENTRY, expected_attributes_1)
+            assert len(ret) > 0, "SIP TunnelMap entry not created"
+            assert len(ret) == 1, "More than 1 SIP TunnMapEntry created"
+            self.mapentry_map[tunnel_name + vidlist[x]] = ret[0]
+            iplinkcmd = "ip link show type vxlan dev " + tunnel_name + "-" + vidlist[x]
+            (exitcode, out) = dvs.runcmd(iplinkcmd)
+            assert exitcode == 0, "Kernel device not created"
+            
+   
+    def check_vxlan_sip_tunnel_delete(self, dvs, tunnel_name):
+        asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+        app_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+
+        tbl = swsscommon.Table(app_db, "VXLAN_TUNNEL_TABLE")
+        status, fvs = tbl.get(self.tunnel_appdb[tunnel_name])
+        assert status == False, "SIP Tunnel entry not deleted"
+
+        tbl = swsscommon.Table(asic_db, self.ASIC_TUNNEL_TERM_ENTRY)
+        status, fvs = tbl.get(self.tunnelterm[tunnel_name])
+        assert status == False, "SIP Tunnel Term entry not deleted"
+
+        tbl = swsscommon.Table(asic_db, self.ASIC_TUNNEL_TABLE)
+        status, fvs = tbl.get(self.tunnel[tunnel_name])
+        assert status == False, "SIP Tunnel entry not deleted"
```

Comment:
line 298 & 306 return same error message, it is difficult to distinguish which error is for what. Please add the db name the log.

Reply:
Done